orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +854,306 @@ def _update_onnx_model_initializers(self, state_tensors):
         for w_i in replace_indices:
             del self._onnx_model.graph.initializer[w_i]
         self._onnx_model.graph.initializer.extend(new_weights)
+
+    def _extract_model_states(self, states):
+        """Extract model states from the training session and load into the states dictionary"""
+
+        model = 'model'
+        model_states = self._training_session.get_model_state(include_mixed_precision_weights=False)
+        states[model] = {}
+
+        # extract trainer model weights from the training session
+        for precision in model_states:
+            states[model][precision] = {}
+            for model_state_key in model_states[precision]:
+                states[model][precision][model_state_key] = torch.from_numpy(model_states[precision][model_state_key])
+
+        # extract untrained (frozen) model weights
+        precision_map = {TensorProto.FLOAT: 'fp32'}
```

Comment:
Is it possible to reuse  `dtype_onnx_to_torch` from _utils.py or create a helper for this there? Hard-coding this kind of information in the middle of the code is error-prone

Reply:
will take a look at that function.

Reply:
Created a new helper function.